ra/registration-authority.go

```diff
@@ -339,6 +347,36 @@ func (ra *RegistrationAuthorityImpl) NewCertificate(req core.CertificateRequest,
 		return emptyCert, err
 	}
 
+	// Check total certificate rate limiting
+	ra.tiMu.RLock()
+	now := ra.clk.Now()
+	if ra.totalIssuedCache == nil || ra.lastIssuedCount.Add(30*time.Minute).Before(now) {
+		// If we don't want reads to block each other we have to life with this horrible
+		// locking grossness (we could possibly use a goroutine that updates the cache
+		// and handles the write locking invisibly to this method but that seems kind
+		// of overkill to be honest)
+		ra.tiMu.RUnlock()
+		ra.tiMu.Lock()
```

Comment:
I think this RUnlock to Lock is likely overkill, but if you're going to do this, you have to check that the ra.totalIssuedCache hasn't been updated (or created) by the time you get the Lock(). If has been, you have to not do this work again.

BTW, I wasn't saying we should do the caching now, just that we would have to at some point and would need a ticket for it.